### PR TITLE
[Sharepoint server] crawl the root site

### PR DESCRIPTION
**Pull request by sjors101**
## Closes https://github.com/elastic/connectors-py/issues/2140

Support crawling and an indexing top-level site collection from Sharepoint server on-premise. Use the existing input field `site_collections` to have the option to define the toplevel (root) site: example input: `foo, bar, /`. Tested agains the Sharepoint 2019 (16.0.0.10337: 1).

#### Pre-Review Checklist
- [x] this PR has a meaningful title
- [x] this PR links to all relevant github issues that it fixes or partially addresses
- [x] if there is no GH issue, please create it. Each PR should have a link to an issue
- [x] this PR has a thorough description
- [x] Covered the changes with automated tests
- [x] Tested the changes locally
- [x] Added a label for each target release version (example: `v7.13.2`, `v7.14.0`, `v8.0.0`)

### Release Note
[Added by Artem]
Make it possible to crawl root site collection for Sharepoint Server connector

### CODE REVIEW

File: connectors/sources/sharepoint_server.py [hunk 1]
```diff
@@ -92,7 +92,12 @@ def __init__(self, configuration):
         self.certificate = self.configuration["ssl_ca"]
         self.ssl_enabled = self.configuration["ssl_enabled"]
         self.retry_count = self.configuration["retry_count"]
-        self.site_collections = self.configuration["site_collections"]
+
+        self.site_collections_path = []
```
Comment (artem-shelkovnikov) [inline connectors/sources/sharepoint_server.py:96]:
```suggestion
        self.site_collection_paths = []
```

I'd probably rename it like this to indicate that it's an array

**Review by artem-shelkovnikov — COMMENTED** [review]
Thank you for your contribution, @sjors101!

Change looks good to me, we'll test it on our side against a real Sharepoint Server instance and will get back to you!

**Comment by artem-shelkovnikov** [comment]
Tested the change, all looks good, thank you for your contribution, @sjors101!

**Comment by artem-shelkovnikov** [comment]
buildkite test this

**Review by artem-shelkovnikov — APPROVED** [review]

**Comment by seanstory** [comment]
@sjors101 be sure to consider joining [our contributor program](https://www.elastic.co/community/contributor)! You can get credit for both the bug you filed and this PR.

As a side note, we'd like to congratulate you, as the first non-Elastic-affiliated contributor to Connectors. If we were a restaurant, we'd pin this code diff to our wall. :) We're very excited about your contribution!

**Comment by sjors101** [comment]
Thanks @seanstory and @artem-shelkovnikov. The connector-framework is developing quite nicely, and the instructions in the contribution guide are great. We use various connectors intensively, expect more contributions coming your way :)

**Pull request merged**